src/ServerRequestFactory.php

```diff
@@ -485,4 +489,34 @@ private static function marshalProtocolVersion(array $server)
 
         return $matches['version'];
     }
+
+    /**
+     * Parse a cookie header according to RFC 6265.
+     *
+     * PHP will replace special characters in cookie names, which results in other cookies not being available due to
+     * overwriting. Thus, the server request should take the cookies from the request header instead.
```

Comment:
We ran into a problem with interacting with a Crowd SSO cookie which is named `crowd.token_key` but was getting set in `$_COOKIE` as `crowd_token_key` and requiring special handling to ensure that those two variants of the string match. This PR would solve that problem.